dashboards-observability/server/adaptors/custom_panels/custom_panel_adaptor.ts

```diff
@@ -71,12 +71,20 @@ export class CustomPanelsAdaptor {
         objectType: 'operationalPanel',
         maxItems: 10000,
       });
-      return response.observabilityObjectList.map((panel: any) => ({
-        name: panel.operationalPanel.name,
-        id: panel.objectId,
-        dateCreated: panel.createdTimeMs,
-        dateModified: panel.lastUpdatedTimeMs,
-      }));
+      const panelList: any[] = [];
+      for (const panel of response.observabilityObjectList) {
+        const panelObject: CustomPanelListType = {
+          name: panel.operationalPanel.name,
+          id: panel.objectId,
+          dateCreated: panel.createdTimeMs,
+          dateModified: panel.lastUpdatedTimeMs
+        }
+        if (panel.operationalPanel.applicationId) {
+          panelObject.applicationId = panel.operationalPanel.applicationId;
+        }
```

Comment:
Does it make sense to filter out the panels here, as we are already iterating through the list? Then we don't need to edit anything in `custom_panels/home.tsx`

Reply:
Oh yea that definitely makes sense, I was thinking maybe we would need to fetch all all panels but I don't think that's ever necessary.